### PR TITLE
Add sample maven/gradle projects.

### PR DESCRIPTION
**Pull request by itohro**

### CODE REVIEW

File: samples/reladomo-gradle-sample/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#Mon Sep 19 14:22:57 JST 2016
```
Comment (gs-rezaem) [inline samples/reladomo-gradle-sample/gradle/wrapper/gradle-wrapper.properties]: needs copyright/license

File: samples/reladomo-gradle-sample/gradlew [hunk 2]
```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env bash
+
```
Comment (gs-rezaem) [inline samples/reladomo-gradle-sample/gradlew]: needs copyright/license

File: samples/reladomo-gradle-sample/gradlew [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+##############################################################################
```
Comment (gs-rezaem) [inline samples/reladomo-gradle-sample/gradlew]: needs copyright/license

File: samples/reladomo-gradle-sample/gradlew.bat [hunk 4]
```diff
@@ -0,0 +1,90 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (gs-rezaem) [inline samples/reladomo-gradle-sample/gradlew.bat]: needs copyright/license

File: samples/reladomo-gradle-sample/src/main/resources/reladomo/db/OBJECT_SEQUENCE.ddl [hunk 5]
```diff
@@ -0,0 +1,8 @@
+drop table if exists OBJECT_SEQUENCE;
```
Comment (gs-rezaem) [inline samples/reladomo-gradle-sample/src/main/resources/reladomo/db/OBJECT_SEQUENCE.ddl]: why are the db objects checked in?

File: samples/reladomo-gradle-sample/src/test/resources/test_data.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class sample.domain.Person
```
Comment (gs-rezaem) [inline samples/reladomo-gradle-sample/src/test/resources/test_data.txt]: needs copyright/license

File: samples/reladomo-maven-sample/src/test/resources/test_data.txt [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class sample.domain.Person
```
Comment (gs-rezaem) [inline samples/reladomo-maven-sample/src/test/resources/test_data.txt]: needs copyright/license

**Comment by itohro** [comment]
You're right, no need to check in the generated ddl/idx files. I removed them and moved to generated-resources folder.

Added missing copyright/license info, too.

**Comment by gs-rezaem** [comment]
The gradle copyright (Copyright 2007-2011 the original author or authors) seems nebulous and I'm not comfortable with that. I think we should not check in the wrapper files. We can add a note that says you must install gradle.

We won't need the notice file or the license files once we remove. Those are only necessary if we copy an artifact or link to (call) their code from our code.

Also, I don't see why we need two directories. The sample application might grow over time (e.g. add a sharding or bitemporal example). I think one directory with both build files is enough.

Finally, the db files are not resources. A resource is something that has to be on the classpath at runtime. I would put the generated files in target/generated-db and for a real project where the generated files are modified (permissions, table spaces, index hints, etc), I would check those in under src/main/db.

**Comment by itohro** [comment]
Thanks for your comments!

- I got rid of the gradle wrapper. Added note about maven/gradle installation requirement in `README.md`, too.

- Regarding notice file, `HelloReladomoApp` calls Reladomo, sl4j and h2. Maybe we don't care Reladome itself but shouldn't we leave sl4j and h2 there? I left them for now. Please let me know.

- I consolidated them into a single sample directly and named `reladomo-sample-simple`. I agree we should add more samples. I'd think it's better to create separate directory for each functionality (e.g. `reladomo-sample-sharding`, `reladomo-sample-bitemporal` etc.) so that it's concise enough for new users to digest what's the minimum requirement for each functionality. We could then add `reladomo-sample-advanced` that has multiple functionalities combined together as more like a real life example.

- I made it to use `target/generated-db` for maven and `build/generated-db `for gradle. For this sample, these still need to be added as resource as the app loads ddl/idx in-memory. I added a commented in `pom.xml`/`build.gradle` to clarify such configuration is not meant to be done for typical production app.

File: samples/reladomo-sample-simple/NOTICE.txt [hunk 8]
```diff
@@ -0,0 +1,20 @@
+Reladomo Maven Sample Project
```
Comment (gs-rezaem) [inline samples/reladomo-sample-simple/NOTICE.txt]: Not just maven anymore!

File: samples/reladomo-sample-simple/src/main/java/sample/HelloReladomoApp.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+        Person personFromDb = Person.findPersonNamed("Taro", "Tanaka");
+
+        logger.info("Hello " + personFromDb.getFirstName() + " " + personFromDb.getLastName());
+    }
+
+    private static void initialize(HelloReladomoApp app)
+    {
+        try
+        {
+            H2ConnectionManager.getInstance().prepareTables();
```
Comment (gs-rezaem) [inline samples/reladomo-sample-simple/src/main/java/sample/HelloReladomoApp.java]: add a note that says this is not normal for prod applications

File: samples/reladomo-sample-simple/src/main/java/sample/HelloReladomoApp.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        }
+    }
+
+    private void initialiseReladomo() throws Exception
+    {
+        try
+        {
+            logger.info("Transaction Timeout is " + MAX_TRANSACTION_TIMEOUT);
+            MithraManager mithraManager = MithraManagerProvider.getMithraManager();
+            mithraManager.setTransactionTimeout(MAX_TRANSACTION_TIMEOUT);
```
Comment (gs-rezaem) [inline samples/reladomo-sample-simple/src/main/java/sample/HelloReladomoApp.java]: add a note that notification should be configured here.

File: samples/reladomo-sample-simple/src/main/java/sample/domain/Person.java [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+public class Person extends PersonAbstract
+{
+	public Person()
+	{
+		super();
+		// You must not modify this constructor. Mithra calls this internally.
+		// You can call this constructor. You can also add new constructors.
+	}
+
```
Comment (gs-rezaem) [inline samples/reladomo-sample-simple/src/main/java/sample/domain/Person.java]: add a domain specific method, like "getFullName()"

File: samples/reladomo-sample-simple/src/main/java/sample/util/H2ConnectionManager.java [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+    {
+        throw new RuntimeException("BulkLoader is not supported");
+    }
+
+    public String getDatabaseIdentifier()
+    {
+        return schemaName;
+    }
+
+    // Load ddl/idx files under generated-db/sql in-memory.
```
Comment (gs-rezaem) [inline samples/reladomo-sample-simple/src/main/java/sample/util/H2ConnectionManager.java]: add a note that not normal for prod

**Comment by itohro** [comment]
Thanks for the quick feedback! Made those changes.

**Pull request merged**